docs/rules/max-statements.md

```diff
@@ -54,16 +90,47 @@ function foo() {
     return 42;
   };
 }
-```
 
-```js
-/*eslint max-statements: ["error", 1, {ignoreTopLevelFunctions: true}]*/  // Maximum of 1 statement.
-(function() {
-  var bar = 1;
+let foo = () => {
+  var foo1 = 1;
+  var foo2 = 2;
+  var foo3 = 3;
+  var foo4 = 4;
+  var foo5 = 5;
+  var foo6 = 6;
+  var foo7 = 7;
+  var foo8 = 8;
+  var foo9 = 9;
+  var foo10 = 10;
   return function () {
+
+    // The number of statements in the inner function does not count toward the
+    // statement maximum.
+
     return 42;
   };
-})()
+}
+```
+
+### ignoreTopLevelFunctions
+
+Examples of additional **correct** code for this rule with the `{ "max": 10 }, { "ignoreTopLevelFunctions": true }` options:
+
+```js
+/*eslint max-statements: ["error", 10, { "ignoreTopLevelFunctions": true }]*/
+function foo() {
```

Comment:
Insert an empty line between eslint comment and first statement.

By the way, despite what I said in max-nested-callbacks, increasing the number of statements here to use the default `{ "max": 10 }` option made these examples seem much more natural.